Add documentation for retry settings for Remote reindex (#6726)
opensearch-project/documentation-website: _im-plugin/reindex-data.md

`socket_timeout` | Time Unit | The wait time for socket reads (default 30s). | No
`connect_timeout` | Time Unit | The wait time for remote connection timeouts (default 30s). | No

> You can use the following cluster settings to configure the retry policy:

Reviewer: ```suggestion
The following table lists the retry policy cluster settings.
```

Setting | Description | Default value
:--- | :--- 

> `reindex.remote.retry.initial_backoff` | Initial backoff time for retries. Subsequent retries will do exponential backoff based on initial backoff time. | 500 ms

Reviewer: ```suggestion
`reindex.remote.retry.initial_backoff` | The initial backoff time for retries. Subsequent retries will follow exponential backoff based on the initial backoff time. | 500 ms
```

> `reindex.remote.retry.max_count` | Max number of retry attempts | 15

Reviewer: ```suggestion
`reindex.remote.retry.max_count` | The maximum number of retry attempts. | 15
```
